### PR TITLE
fix(skipUntil): fix skipUntil when innerSubscription is null

### PR DESCRIPTION
**Pull request by claylaut**
**Description:** Fix a minor issue with skipUntil

**Related issue:**
https://github.com/ReactiveX/rxjs/issues/3685

### CODE REVIEW

**Comment by cartant** [comment]
Your fix looks appropriate, to me, but, ideally, the PR should also include a test that's based on code that reproduces the error.

The problem is effected because the implementation doesn't account for the synchronous completion of the notifier. You don't need to use a `BehaviorSubject` to effect the error; a simple, synchronous observable will do:

```ts
import { of } from "rxjs";
import { skipUntil } from "rxjs/operators";

of(1).pipe(
  skipUntil(of(2))
).subscribe(x => console.log(x));
```

**Comment by claylaut** [comment]
@cartant I've just simplified my usage example and opened an issue. 
I've tested the fix by modifying the node modules and it worked fine. Unfortunately, without this fix, I have to revert to 5.5.10.

I m unsure how to do the test if you can guide me I will appreciate it.

**Comment by claylaut** [comment]
@cartant thanks a lot for your help. 
I've updated the tests and they passed successfully.
PR Updated.

File: src/internal/operators/skipUntil.ts [hunk 1]
```diff
@@ -57,7 +57,9 @@ class SkipUntilSubscriber<T, R> extends OuterSubscriber<T, R> {
              outerIndex: number, innerIndex: number,
              innerSub: InnerSubscriber<T, R>): void {
     this.hasValue = true;
-    this.innerSubscription.unsubscribe();
+    if(this.innerSubscription) {
```
Comment (martinsik) [inline src/internal/operators/skipUntil.ts]: This is just nitpicking but in all code around here there's always space between `if` and the starting parenthesis `if (this.innerSubscription)`.

Comment (claylaut) [reply src/internal/operators/skipUntil.ts]: fixed

**Comment by cartant** [comment]
@claylaut The repository's maintainers will eventually get around to reviewing and, likely, merging your PR. You will just need to be a little patient, as they're busy people with other priorities. In the interim, you could use a scheduler for a temporary workaround:

```ts
import { asapScheduler, of } from "rxjs";
import { skipUntil } from "rxjs/operators";

of(1).pipe(
  skipUntil(of(2, asapScheduler))
).subscribe(x => console.log(x));
```

**Comment by benlesh** [comment]
Thanks, @claylaut!

**Comment by claylaut** [comment]
@benlesh thanks for merge.

@cartant thanks for the workaround 👍

**Pull request merged**